### PR TITLE
Use persistent_term for faster adapter config lookups

### PR DESCRIPTION
**Pull request by josevalim**

### CODE REVIEW

File: lib/ecto/repo/registry.ex [hunk 1]
```diff
@@ -18,9 +18,8 @@ defmodule Ecto.Repo.Registry do
   end
 
   def lookup(repo) when is_atom(repo) do
-    GenServer.whereis(repo)
-    |> Kernel.||(raise "could not lookup Ecto repo #{inspect repo} because it was not started or it does not exist")
-    |> lookup()
+    :persistent_term.get(repo) ||
```
Comment (wojtekmach) [inline lib/ecto/repo/registry.ex:21]: Double checking: I think get/1 either returns :undefined or raises so wondering if we should use `get(repo, nil)` instead. If we are setting it prior, nevermind!

Comment (josevalim) [reply lib/ecto/repo/registry.ex:21]: Yes, that's what we wanted, thank you!!!

Comment (josevalim) [reply lib/ecto/repo/registry.ex:21]:
```suggestion
    :persistent_term.get(repo, nil) ||
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**